client/src/components/NotificationPopUp/NotificationPopUp.tsx

```diff
@@ -0,0 +1,161 @@
+import { useState, useEffect } from 'react';
+import { Link } from 'react-router-dom';
+import useStyles from './useStyles';
+import NotificationsNoneIcon from '@material-ui/icons/NotificationsNone';
+import Badge from '@material-ui/core/Badge';
+import Popover from '@material-ui/core/Popover';
+import Typography from '@material-ui/core/Typography';
+import Button from '@material-ui/core/Button';
+import Grid from '@material-ui/core/Grid';
+import Avatar from '@material-ui/core/Avatar';
+import { withStyles } from '@material-ui/core/styles';
+import { getNotification, updateNotification } from '../../helpers/APICalls/notification';
+import { Notification } from '../../interface/User';
+
+const GlobalCss = withStyles({
+  // @global is handled by jss-plugin-global.
+  '@global': {
+    // You should target [class*="MuiButton-root"] instead if you nest themes.
+    '.MuiPopover-paper': {
+      paddingBottom: 40,
+    },
+    '*::-webkit-scrollbar': {
+      width: '0px',
+    },
+  },
+})(() => null);
+
+const NotificationPopUp = (): JSX.Element => {
+  const [notifications, setNotifications] = useState<Notification[]>([]);
+  const classes = useStyles();
+  const [anchorEl, setAnchorEl] = useState<HTMLButtonElement | null>(null);
+
+  const open = Boolean(anchorEl);
+  const id = open ? 'simple-popover' : undefined;
+
+  const handleClose = () => {
+    setAnchorEl(null);
+  };
+
+  useEffect(() => {
```

Comment:
Currently we are fetching all the notifications each time this component mounts, maybe consider using a context (new/existing) as it would improve performance?

Reply:
made changes in the recent commit